pandas/tests/io/pytables/test_put.py

```diff
@@ -194,10 +199,14 @@ def test_put_compression_blosc(setup_path):
             store.put("b", df, format="fixed", complib="blosc")
 
         store.put("c", df, format="table", complib="blosc")
-        tm.assert_frame_equal(store["c"], df)
+        expected = df
+        if using_infer_string:
+            expected.columns = expected.columns.astype("str")
```

Comment:
Do we always need to cast to the "str" type? I was hoping we could retain the same data type that we started with; not sure of what limitations pytables may impose

Reply:
The current implementation writes out bytes and then these are inferred upon being read. I believe that PDEP-14 states such inference should resolve as using non-object.

> For pandas 3.0, a "str" string dtype is enabled by default, i.e. this string dtype will be used as the default dtype for text data when creating pandas objects (e.g. inference in constructors, I/O functions).

To make this not use inference I think would be an additional enhancement on top of the existing implementation.

Reply:
Oh we are talking about two different things. If you remove the `dtype=object` from the other conversation with should round trip to a string type. What I've seen on other PRs though is that it can be easy to mix up str versus string. I assumed that's what this cast was trying to work around